cmd/kube-proxy/app/server.go

```diff
@@ -168,6 +168,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.Var(utilflag.IPVar{Val: &o.config.BindAddress}, "bind-address", "The IP address for the proxy server to serve on (set to '0.0.0.0' for all IPv4 interfaces and '::' for all IPv6 interfaces)")
 	fs.Var(utilflag.IPPortVar{Val: &o.config.HealthzBindAddress}, "healthz-bind-address", "The IP address with port for the health check server to serve on (set to '0.0.0.0:10256' for all IPv4 interfaces and '[::]:10256' for all IPv6 interfaces). Set empty to disable.")
 	fs.Var(utilflag.IPPortVar{Val: &o.config.MetricsBindAddress}, "metrics-bind-address", "The IP address with port for the metrics server to serve on (set to '0.0.0.0:10249' for all IPv4 interfaces and '[::]:10249' for all IPv6 interfaces). Set empty to disable.")
+	fs.BoolVar(&o.config.MetricsBindHardFail, "metrics-bind-hard-fail", o.config.MetricsBindHardFail, "If true kube-proxy will treat metrics bind failure as fatal and exit")
```

Comment:
Should we cover both metrics and healthz?  E.g. `--bind-address-hard-fail`?

Reply:
Thank you very much @thockin ! I'll try to refactor my code.